src/mpid/common/sched/mpidu_sched.c

```diff
@@ -490,13 +488,9 @@ int MPIDU_Sched_start(MPIR_Sched_t * sp, MPIR_Comm * comm, int tag, MPIR_Request
 
     /* finally, enqueue in the list of all pending schedules so that the
      * progress engine can make progress on it */
-    if (all_schedules.head == NULL) {
-        mpi_errno = MPID_Progress_register_hook(MPIDU_Sched_progress, &nbc_progress_hook_id);
-        if (mpi_errno)
-            MPIR_ERR_POP(mpi_errno);
+    if (all_schedules.head == NULL)
```

Comment:
Is there any chance we have to worry about thread safety here?

Reply:
Not for the global lock case.  Even in the workq model that we are working on, the progress engine is always protected with a lock.  So, I believe, this will be correct either way.

Reply:
Ok. That sounds good then.